File: tests/cases/lock/locked/python/reformat/inputs/src/orders/billing.py
from cdec_rules import locked


class Receipt:
    """A brand new class docstring, which is not part of the locked body."""

    def __init__(self, total: float, lines: int) -> None:
        self.total: float = total
        self.lines: int = lines

    def unrelated_new_method(self) -> None:
        """Inserted ABOVE the locked method - a lock is an AST identity,
        not a line range."""

    @locked(reason="receipt wording is contractual; finance signed off on it")
    def formatted(
        self,
    ) -> str:
        # a new explanatory comment
        return f"{self.lines} line(s) - total {self.total:.2f}"

File: tests/cases/lock/locked/csharp/reformat/inputs/src/Orders/Billing.cs
using CodeConstraints.Rules;

namespace Orders;

public class Receipt
{
    public decimal Total { get; }
    public int Lines { get; }

    public Receipt(decimal total, int lines) { Total = total; Lines = lines; }

    // Inserted ABOVE the locked method - a lock is an AST identity, not a
    // line range.
    public void UnrelatedNewMethod() { }

    [Locked(Reason = "receipt wording is contractual; finance signed off on it")]
    public string Formatted()
    {
        // a new explanatory comment
        return $"{Lines} line(s) - total {Total}";
    }
}
